nntrainer/layers/split_layer.cpp

```diff
@@ -26,29 +26,41 @@ static constexpr size_t SINGLE_INOUT_IDX = 0;
 SplitLayer::SplitLayer() :
   Layer(),
   leading_helper_dim(1),
-  split_props(props::SplitDimension()) {}
+  split_props(props::SplitDimension(), props::SplitNumber()) {}
 
 void SplitLayer::finalize(InitLayerContext &context) {
   NNTR_THROW_IF(context.getNumInputs() != 1, std::invalid_argument)
     << "Error: only a single input is supported with split layer";
 
   unsigned int split_dimension = std::get<props::SplitDimension>(split_props);
 
+  if (std::get<props::SplitNumber>(split_props).empty()) {
+    std::get<props::SplitNumber>(split_props)
+      .set(context.getNumRequestedOutputs());
+  }
+  unsigned int split_number = std::get<props::SplitNumber>(split_props);
+
   /**
    * The split is only done along the split_dimension dimension.
-   * For example, consider input dimension [b,c,h,w],
-   * 1. axis = 1, output_dim = [b,1,h,w], num_outputs = c
-   * 2. axis = 2, output_dim = [b,c,1,w], num_outputs = h
-   * 3. axis = 3, output_dim = [b,c,h,1], num_outputs = w
+   * For example, consider input dimension [b,c,h,w], split_number = n
+   * 1. axis = 1, output_dim = [b,n,h,w], num_outputs = c//n
+   * 2. axis = 2, output_dim = [b,c,n,w], num_outputs = h//n
+   * 3. axis = 3, output_dim = [b,c,h,n], num_outputs = w//n
```

Comment:
Could you add this property expect the data is continuous?

Reply:
Added a comment split layer assumes input data is continuous